contracts/VaultFactory.vy

```diff
@@ -24,27 +20,53 @@ event UpdateProtocolFeeRecipient:
     old_fee_recipient: indexed(address)
     new_fee_recipient: indexed(address)
 
+event UpdateCustomProtocolFee:
+    vault: indexed(address)
+    new_custom_protocol_fee: uint16
+
+event RemovedCustomProtocolFee:
+    vault: indexed(address)
+
+event FactoryShutdown():
+    pass
+
 event UpdateGovernance:
     governance: indexed(address)
 
 event NewPendingGovernance:
     pending_governance: indexed(address)
 
 struct PFConfig:
-  fee_bps: uint16
-  fee_last_change: uint32
-  fee_recipient: address
+    # Percent of protocol's split of fees in Basis Points.
+    fee_bps: uint16
+    # Address for protocol fees to get paid to.
+    fee_recipient: address
 
-MAX_FEE_BPS: constant(uint16) = 25 # max protocol management fee is 0.25% annual
-API_VERSION: constant(String[28]) = "3.1.0"
+# The max amount the protocol fee can be set to.
+MAX_FEE_BPS: constant(uint16) = 5_000
+# Identifier for this version of the vault.
+API_VERSION: constant(String[28]) = "3.0.1-beta"
```

Comment:
You mean instead of the "-beta"?

Im just using it for pre-prod versions for versioning for the different audit stages, and so prod versions can start back at 3.0.0. With no worsd